src/pages/calculator/settings/index.tsx

```diff
@@ -1,8 +1,48 @@
+import React, { useCallback } from "react"
+
 import CalculatorLayout from "@/components/layouts/calculator"
 import { ComponentWithLayout } from "@/components/layouts/types"
+import { useAppDispatch, useAppSelector } from "@/store/hooks"
+import {
+  selectTravelerGender,
+  setTravelerGender,
+  TravelerGender,
+} from "@/store/settings/settingsSlice"
+import { Switch } from "@headlessui/react"
 
 export const SettingsPage: React.FC & ComponentWithLayout = () => {
-  return <div className="flex justify-center w-full max-w-5xl">Settings</div>
+  const dispatch = useAppDispatch()
+  // const [twin, setTwin] = useState<TravelerGender>(useAppSelector(selectTravelerGender))
+  const twin: TravelerGender = useAppSelector(selectTravelerGender)
+
+  const setTwin = useCallback(
+    (gender) => {
+      dispatch(setTravelerGender(gender))
+    },
+    [dispatch],
+  )
+
+  return (
+    <Switch.Group as="div" className="flex space-x-4">
+      <Switch.Label>Select twin. Currently: {twin}</Switch.Label>
+      <Switch
+        as="button"
+        checked={twin === "female"}
+        onChange={() => setTwin(twin === "male" ? "female" : "male")}
```

Comment:
Let's actually extract this inline `() => setTwin...` call into a single useCallback at the top (and inline the setTwin definition inside instead):

something like
```typescript
const toggleTwin = useCallback(() => {
	const swap = twin === "male" ? "female" : "male"
	dispatch(setTravelerGender(swap))
}, [twin, dispatch])
```

The reason is because on every component re-render, React will create an entirely new anonymous function call for `onChange={() => setTwin...`. If we switch it to `onChange={toggleTwin}`, the toggleTwin function is created once and then memoized by useCallback so subsequent re-renders will not create an entirely new anonymous function call.